File: killrvideo-service-statistics/src/main/java/com/killrvideo/service/statistic/dao/VideoPlaybackStatsDao.java
package com.killrvideo.service.statistic.dao;

import com.datastax.oss.driver.api.core.MappedAsyncPagingIterable;
import com.datastax.oss.driver.api.mapper.annotations.*;
import com.killrvideo.service.statistic.dto.VideoPlaybackStats;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

@Dao
public interface VideoPlaybackStatsDao {
    @Select(customWhereClause = "videoid in :listOfVideoIds")
    @StatementAttributes(consistencyLevel = "LOCAL_ONE")
    CompletableFuture<MappedAsyncPagingIterable<VideoPlaybackStats>> getNumberOfPlays(@CqlName("listOfVideoIds") List<UUID> listOfVideoIds);
}

File: killrvideo-service-statistics/src/main/kotlin/com/killrvideo/service/statistic/dao/VideoPlaybackStatsDao.kt
package com.killrvideo.service.statistic.dao

import com.datastax.oss.driver.api.core.MappedAsyncPagingIterable
import com.datastax.oss.driver.api.mapper.annotations.*
import com.killrvideo.service.statistic.dto.VideoPlaybackStats
import java.util.*
import java.util.concurrent.CompletableFuture

@Dao
interface VideoPlaybackStatsDao {
    @Select(customWhereClause = "videoid in :listOfVideoIds")
    @StatementAttributes(consistencyLevel = "LOCAL_ONE")
    fun getNumberOfPlays(@CqlName("listOfVideoIds") listOfVideoIds: List<UUID>):
            CompletableFuture<MappedAsyncPagingIterable<VideoPlaybackStats>>

    @Increment(entityClass = [VideoPlaybackStats::class])
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    fun increment(videoid: UUID, views: Long): CompletableFuture<Void>
}
